services/src/main/java/io/scalecube/services/Microservices.java

```diff
@@ -244,11 +258,37 @@ public ServiceDiscovery discovery() {
    */
   public Mono<Void> shutdown() {
     return Mono.defer(
-        () ->
-            Mono.whenDelayError(
-                discoveryBootstrap.shutdown(),
-                gatewayBootstrap.shutdown(),
-                transportBootstrap.shutdown()));
+        () -> {
+          shutdown.onComplete();
+          return onShutdown;
```

Comment:
if we have the separate method, why do we need to return anything? I propose to return just `void` from this method.

Reply:
Because I don't want to change clients all around the code.